examples/specialty_plots/ishikawa.py

```diff
@@ -0,0 +1,232 @@
+
+        # calculate spine spacing
+        # calculate couple pairs, at least 1 pair to start at middle branch
+        pairs = len(list(data.keys())) // 2 + len(list(data.keys())) % 2 + 1
+        spacing = plen / pairs
+
+        spine_count = 0
+        s = spacing
+        # draw spine
+        for k in data.keys():
+            # calculate arrow position in the graph
+            # fix primary spines spacing
+            if reclevel == 1:
+                if len(primary_spines_rel_xpos) == len(list(data.keys())):
+                    x_end = primary_spines_rel_xpos[spine_count]
+                else:
+                    x_end = xpe - (s / 1.5) * np.cos(np.radians(palpha))
+                y_end = ype - s * np.sin(np.radians(palpha))
+
+                x_start = x_end - s * np.cos(np.radians(alpha))
+                y_start = y_end - s * np.sin(np.radians(alpha))
+            else:
+                x_end = xpe - s * np.cos(np.radians(palpha))
+                y_end = ype - s * np.sin(np.radians(palpha))
+
+                x_start = x_end - s * np.cos(np.radians(alpha))
+                y_start = y_end - s * np.sin(np.radians(alpha))
+            if debug:
+                _log.debug(
+                    f'k: {k}, x_start: {x_start}, y_start: {y_start}, '
+                    f'x_end: {x_end} y_end: {y_end}, alpha: {alpha} \n '
+                    f'reclevel: {reclevel}, '
+                    f's:{s}, plen:{plen}, palpha:{palpha}')
+
+            # draw arrow arc
+            if reclevel == 1:
+                props = dict(boxstyle='round', facecolor='lightsteelblue', alpha=1.0)
+                spine = ax.annotate(str.upper(k), xy=(x_end, y_end),
+                                    xytext=(x_start, y_start),
+                                    arrowprops=dict(arrowstyle="->",
+                                                    facecolor=spine_color),
+                                    bbox=props, weight='bold')
+            else:
+                props = dict(boxstyle='round', facecolor='lavender', alpha=1.0)
+                spine = ax.annotate(k, xy=(x_end, y_end), xytext=(x_start, y_start),
+                                    arrowprops=dict(arrowstyle="->",
+                                                    facecolor=spine_color),
+                                    bbox=props)
+            # Call recursion to draw subspines
+            drawspines(data=data[k], ax=ax, parentspine=spine,
+                       alpha_ps=alpha_ps, alpha_ss=alpha_ss,
+                       spine_color=spine_color, reclevel=reclevel + 1,
+                       max_recursion_level=max_recursion_level)
+            # no primary_spines_rel_xpos is needed to be passed on recursion
+            # next spine settings - same level
+            alpha *= -1
+            spine_count += 1
+            if spine_count % 2 == 0:
+                s = s + spacing
+    return None
+
+
+def ishikawaplot(data, figsize=(20, 10), left_margin: float = 0.05,
```

Comment:
This method really just acts on a single axes.  Strongly suggest re-structuring without `figsize` and just passing in an axes.  If no axes is passed in, then make a new figure, and return the new axes.